x-pack/plugins/upgrade_assistant/server/lib/reindexing/reindex_service.ts

```diff
@@ -317,7 +321,10 @@ export const reindexServiceFactory = (
   const startReindexing = async (reindexOp: ReindexSavedObject) => {
     const { indexName, reindexOptions } = reindexOp.attributes;
 
-    if (reindexOptions?.openAndClose === true) {
+    const indicesState = await esIndicesStateCheck(callAsUser, [indexName]);
```

Comment:
Is it worth adding a comment here to reflect your thoughts in the PR description? For example:

```js
// Where possible, derive reindex options at the last moment before reindexing
// to prevent them from becoming stale as they wait in the queue.
```